synthesizer/src/vm/verify.rs

```diff
@@ -39,6 +39,7 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
         transaction: &Transaction<N>,
         rejected_id: Option<Field<N>>,
         rng: &mut R,
+        already_verified: bool,
```

Comment:
The addition of the `already_verified` flag feels unsafe. This make it incredibly easy to accidentally or intentionally skip a transaction verification.

Reply:
I feel like this conditional logic should only exist in `check_transaction_basic`, and not low level in the VM.

Reply:
@raychu86 hmm, how about using type safety instead? i.e. have a wrapper struct like `VerifiedTransaction { transaction }` or using generics like `Transaction<N: Network, V: VerificationStatus>`?